src/core/account/account.ts

```diff
@@ -0,0 +1,252 @@
+ * use the `SingleKey` authentication scheme.
+ */
+export interface CreateEd25519SingleKeyAccountFromPrivateKeyArgs {
+  privateKey: Ed25519PrivateKey;
+  address?: AccountAddressInput;
+  legacy: false;
+}
+
+/**
+ * Arguments for creating an `SingleKeyAccount` from any supported private key
+ * that is not an `Ed25519PrivateKey`.
+ * The `legacy` argument defaults to false and cannot be explicitly set to true.
+ */
+export interface CreateSingleKeyAccountFromPrivateKeyArgs {
+  privateKey: Exclude<PrivateKey, Ed25519PrivateKey>;
+  address?: AccountAddressInput;
+  legacy?: false;
+}
+
+/**
+ * Arguments for creating an opaque `Account` from any supported private key.
+ * This is used when the private key type is not known at compilation time.
+ */
+export interface CreateAccountFromPrivateKeyArgs {
+  privateKey: PrivateKey;
+  address?: AccountAddressInput;
+  legacy?: boolean;
+}
+
+/**
+ * Arguments for generating an `Ed25519Account`.
+ * This is the input type used by default.
+ */
+export interface GenerateEd25519AccountArgs {
+  scheme?: SigningSchemeInput.Ed25519;
+  legacy?: true;
+}
+
+/**
+ * Arguments for generating an `SingleKeyAccount` with ah underlying `Ed25519PrivateKey`.
+ * The `legacy` argument needs to be explicitly set to false,
+ * otherwise an `Ed25519Account` will be returned instead.
+ */
+export interface GenerateEd25519SingleKeyAccountArgs {
+  scheme?: SigningSchemeInput.Ed25519;
+  legacy: false;
+}
+
+/**
+ * Arguments for generating an `SingleKeyAccount` with any supported private key
+ * that is not an `Ed25519PrivateKey`.
+ * The `legacy` argument defaults to false and cannot be explicitly set to true.
+ */
+export interface GenerateSingleKeyAccountArgs {
+  scheme: Exclude<SigningSchemeInput, SigningSchemeInput.Ed25519>;
+  legacy?: false;
+}
+
+/**
+ * Arguments for generating an opaque `Account`.
+ * This is used when the input signature scheme is not known at compilation time.
+ */
+export interface GenerateAccountArgs {
+  scheme?: SigningSchemeInput;
+  legacy?: boolean;
+}
+
+/**
+ * Arguments for deriving a private key from a mnemonic phrase and a BIP44 path.
+ */
+export interface PrivateKeyFromDerivationPathArgs {
+  path: string;
+  mnemonic: string;
+}
+
+/**
+ * Interface for a generic Aptos account.
+ *
+ * The interface is defined as abstract class to provide a single entrypoint for account generation,
+ * either through `Account.generate()` or `Account.fromDerivationPath`.
+ * Despite this being an abstract class, it should be treated as an interface and enforced using
+ * the `implements` keyword.
+ *
+ * Note: Generating an account instance does not create the account on-chain.
+ */
+export abstract class Account {
+  /**
+   * Private key associated with the account.
+   * Note: this will be removed in the next major release,
+   *  as not all accounts have a private key.
+   */
+  abstract readonly privateKey: PrivateKey;
```

Comment:
Mmm good point, I also thought about it.
Although it would be `CryptoPrivateKey` as it would not be associated to a specific authentication scheme.
The only possible private keys are `Ed25519PrivateKey` and `Secp256k1PrivateKey`

Reply:
Passkeys add `Secp256r1` -- its just not included yet, there's a draft sdk PR for it though -- and ZkID may add other schemes. I personally would like it to be consistent just so its easier to find things

Reply:
Yes that makes sense